### PR TITLE
chore: flip back from pinned and resolved versions of CLI everywhere

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Unpinning CLI packages, and pushing to run E2E flows as well, will continue iterating to address remaining failing tests after that pass is complete.

Changes and verifications:
- [x] Remove CI and Target steps targeting pinned Version
- [x] Use `yarn update-cli-packages` to fix all nested `package.json` files (and ensure we're back on caret deps).
- [x] Verify that `yarn setup-dev` creates a functional `amplify-dev` target for testing and debugging.
- [x] Verify that `yarn cloud-e2e` runs an e2e suite and is functionality.
- [x] Fix any newly failing e2e tests.
- [x] Address specifically the `api_rest_lambda_migration_v10` test that has been known to fail.

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A

#### Description of how you validated changes
Build, Unit, E2E Tests

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alharris-at** [comment]
Current state of E2Es

```
Error:

🛑 Inaccessible host: `amplify.ap-northeast-3.amazonaws.com' at port `undefined'. This service may not be available in the `ap-northeast-3' region.

Tests:

/src/__tests__/migration/api.key.migration5.test.ts
	amplify add api init project, run invalid migration trying to add more than one gsi, and check for error
	amplify add api init project, run invalid migration when adding more than one gsi on the same table
/src/__tests__/schema-auth-12.test.ts
	api directives @auth batch 2 > auth owner7
	api directives @auth batch 2 > auth ownerMultiAuthRules
/src/__tests__/schema-iterative-rollback-2.test.ts
	Iterative Rollback - removing two @keys > should support rolling back from the 2nd de...
/src/__tests__/schema-auth-5.test.ts
	api directives @auth batch 5 > auth usingOidc
	api directives @auth batch 5 > auth customClaims
	api directives @auth batch 5 > auth combiningAuthRules1
	api directives @auth batch 5 > auth combiningAuthRules2

Error:

🛑 The following resources failed to deploy:
	Resource Name: OpenSearchDomain (AWS::Elasticsearch::Domain)
	Event Type: create
	Reason: Invalid instance type: t2.small.elasticsearch (Service: AWSElasticsearch; Status Code: 409; Error Code: InvalidTypeException; Request ID: c92fb31c-0a44-42a2-a63f-785c9742798c; Proxy: null)

Tests:

/src/__tests__/graphql-v2/searchable-node-to-node-encryption/searchable-previous-deployment-no-node-to-node.test.ts
	searchable deployment when previous deployed state had node to node encryption disabl...

Error:

CDK Credential/Bootstrap issue, and maybe mismatched bucket account/region (I'm wondering if we fell back to root for the last test)

Tests:

/src/__tests__/all-auth-modes.test.ts
	CDK Auth Modes > CDK deploys with all auth modes enabled

Error:

That one snapshot issue :(

Tests:

/src/__tests__/migration_tests_v10/api-rest-lambda.migration.test.ts
	api lambda migration tests > ...adds lambda with multiple rest APIs with v10 and pull...
```

Currently looking into that ap-northeast-3 thing, while I rerun (hoping the CDK auth modes was triggered by a timeout and fallback to a different region or something).

**Comment by alharris-at** [comment]
Update, a few transient failures were resolved on re-run, but elasticsearch version issue, ap-northeast-3 issue, and failing existing test are still failing.

**Comment by alharris-at** [comment]
Only the previously failing test is still failing now.

**Comment by alharris-at** [comment]
To be discussed w/ the team tomorrow, but I'm just removing the duplicate test suite from Amplify CLI

File: scripts/update-cli-dependencies.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
...
...amplify-category-notifications @aws-amplify/amplify-category-predictions amplify-cli-core @aws-amplify/amplify-cli-core @aws-amplify/amplify-cli-logger amplify-cli-shared-interfaces amplify-console-hosting amplify-container-hosting amplify-dotnet-function-runtime-provider amplify-dotnet-function-template-provider amplify-dynamodb-simulator amplify-e2e-tests @aws-amplify/amplify-frontend-android @aws-amplify/amplify-frontend-flutter @aws-amplify/amplify-frontend-ios @aws-amplify/amplify-frontend-javascript amplify-function-plugin-interface amplify-go-function-runtime-provider @aws-amplify/amplify-go-function-template-provider amplify-graphql-migration-tests amplify-headless-interface amplify-java-function-runtime-provider amplify-java-function-template-provider amplify-nodejs-function-runtime-provider @aws-amplify/amplify-nodejs-function-template-provider amplify-prompts @aws-amplify/amplify-prompts @aws-amplify/amplify-provider-awscloudformation amplify-python-function-runtime-provider @aws-amplify/amplify-python-function-template-provider amplify-storage-simulator amplify-util-headless-input @aws-amplify/amplify-util-import @aws-amplify/amplify-util-mock amplify-velocity-template"
```
Comment (alharris-at) [inline scripts/update-cli-dependencies.sh:7]: Removed unneeded migration-tests from this list (since we don't depend on it)

**Review by palpatim — APPROVED** [review]

**Review by AaronZyLee — APPROVED** [review]

**Pull request merged**